vital/range/iterator_range.h

```diff
@@ -29,6 +32,10 @@ class iterator_range
   Iterator begin() const { return m_begin; }
   Iterator end() const { return m_end; }
 
+  bool empty() const { return !( m_begin != m_end ); }
```

Comment:
This seems like the double-negative way of saying `return ( m_begin == m_end )`.  Functionally, I would think it's the same, is it not?

Reply:
Yes, but I've seen custom iterators in KWIVER (e.g. `vital/range/indirect.h:34`) that only implement the `!=` operator. Unsure whether it is a good idea to encourage / enable this behavior or not. Probably should just use `==` and force all iterators to implement it? But I did it like this in the PR so the topic would come up and I could get input.

Reply:
Well, that's a good reason to use `!=`.  But I agree that `==` should be encouraged.  My preference is for `==` if it exists, but if it's not available, `!-` may be necessary because fixing the rest of Kwiver is probably out of scope.

Reply:
...but that should only be an issue when those iterators actually get used. If it builds *now*, "broken" iterators can be fixed when someone goes to try to use `iterator_range::empty` with such iterators; we don't have to try to fix them all *now*.

I don't have a strong opinion, just pointing out that "fixing the rest of Kwiver is probably out of scope" isn't necessarily a (strong) argument against `==`.

Reply:
FWIW, C++17 seems to prioritize `==`. C++20 more explicitly specifies [`equality_comparable`](https://timsong-cpp.github.io/cppwp/n4861/concept.equalitycomparable), which requires *both* `==` and `!=`. So there seems to be a good argument that not implementing `==` is "broken".